releasenotes/notes/add-random-generators-9f99e57b5e4188f2.yaml

```diff
@@ -0,0 +1,16 @@
+---
+features:
+  - |
+    Three new random graph generators, ``gnp_random_graph``, ``gnm_random_graph``
+    and ``random_geometric_graph``, have been added to the ``rustworkx-core``
+    crate in the ``generators`` module. The ``gnp_random_graph`` takes inputs of
+    the number of nodes and a probability for adding edges. The ``gnp_random_graph``
+    takes inputs of the number of nodes and number of edges. The
+    ``random_geometric_graph`` creates a random graph within an n-dimensional
+    cube.
+upgrade:
+  - |
+    Passing a negative value to the ``probability`` argument to the
+    :func:`~rustworkx.gnp_directed_random_graph` or the
+    :func:`~rustworkx.gnp_undirected_random_graph` function will now cause
+    an OverflowError to be raised.
```

Comment:
```suggestion
    an ``OverflowError`` to be raised. Previously, a ``ValueError`` would be raised in this situation.
    This was changed to be consistent with other similar error conditions in other functions in
    the library.
```

Reply:
Done in https://github.com/Qiskit/rustworkx/pull/818/commits/363fb65950a7685ae302929fb5141eba7b22a125. Also fixed the `token_swapper` test failure that was using `gnm_random_graph`.